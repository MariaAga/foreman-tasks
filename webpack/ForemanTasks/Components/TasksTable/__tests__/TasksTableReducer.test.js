import { testReducerSnapshotWithFixtures } from 'react-redux-test-utils';
import {
  TASKS_TABLE_ID,
  TASKS_TABLE_SET_SORT,
  TASKS_TABLE_SET_PAGINATION,
} from '../TasksTableConstants';
import reducer from '../TasksTableReducer';

const fixtures = {
  'should return the initial state': {},
  'should handle TASKS_TABLE_SUCCESS': {
    action: {
      type: `${TASKS_TABLE_ID}_SUCCESS`,
      payload: {
        subtotal: 120,
        page: 3,
        per_page: 12,
      },
    },
  },
  'should handle TASKS_TABLE_SET_SORT': {
    action: {
      type: TASKS_TABLE_SET_SORT,
      payload: { by: 'a', order: 'b' },
    },
  },

  'should handle TASKS_TABLE_SET_PAGINATION': {
    action: {
      type: TASKS_TABLE_SET_PAGINATION,
      payload: { page: 4, perPage: 7 },
    },
  },
};

describe('TasksTablePaginationReducer reducer', () =>
  testReducerSnapshotWithFixtures(reducer, fixtures));
