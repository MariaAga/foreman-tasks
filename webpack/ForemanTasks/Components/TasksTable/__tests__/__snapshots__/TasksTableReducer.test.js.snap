// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TasksTablePaginationReducer reducer should handle TASKS_TABLE_SET_PAGINATION 1`] = `
Object {
  "tasksTableQuery": Object {
    "pagination": Object {
      "page": 4,
      "perPage": 7,
    },
  },
}
`;

exports[`TasksTablePaginationReducer reducer should handle TASKS_TABLE_SET_SORT 1`] = `
Object {
  "tasksTableQuery": Object {
    "sort": Object {
      "by": "a",
      "order": "b",
    },
  },
}
`;

exports[`TasksTablePaginationReducer reducer should handle TASKS_TABLE_SUCCESS 1`] = `
Object {
  "tasksTableQuery": Object {
    "itemCount": 120,
    "pagination": Object {
      "page": 3,
      "perPage": 12,
    },
  },
}
`;

exports[`TasksTablePaginationReducer reducer should return the initial state 1`] = `
Object {
  "tasksTableQuery": Object {},
}
`;
